tests/integration/test_batch_predictions.py

```diff
@@ -0,0 +1,111 @@
+#
+# Copyright 2024 DataRobot, Inc. and its affiliates.
+#
+# All rights reserved.
+#
+# DataRobot, Inc.
+#
+# This is proprietary source code of DataRobot, Inc. and its
+# affiliates.
+#
+# Released under the terms of DataRobot Tool and Utility Agreement.
+# https://www.datarobot.com/wp-content/uploads/2021/07/DataRobot-Tool-and-Utility-Agreement.pdf
+
+import numpy as np
+import pandas as pd
+import pytest
+
+from datarobotx.idp.batch_predictions import get_update_or_create_batch_prediction_job
+from datarobotx.idp.datasets import get_or_create_dataset_from_df
+from datarobotx.idp.use_cases import get_or_create_use_case
+
+
+@pytest.fixture
+def df():
+    return pd.DataFrame(np.random.randint(0, 100, size=(20, 4)), columns=list("ABCD"))
+
+
+@pytest.fixture
+def use_case(dr_endpoint, dr_token, cleanup_dr):
+    with cleanup_dr("useCases/"):
+        yield get_or_create_use_case(
+            dr_endpoint,
+            dr_token,
+            "pytest use case",
+        )
+
+
+# Using pytest fixture dataset...
+@pytest.fixture
+def dataset(dr_endpoint, dr_token, df, use_case, cleanup_dr):
+    with cleanup_dr("datasets/", id_attribute="datasetId"):
+        yield get_or_create_dataset_from_df(
+            dr_endpoint,
+            dr_token,
+            use_case_id=use_case,
+            name="pytest_test_autopilot_dataset",
+            data_frame=df,
+        )
+
+
+@pytest.fixture
+def deployment_id():
+    """Random deployment ID."""
+    return "6671e1c5a536861654e445a3"
```

Comment:
Oh if only it were so easy to just choose a deployment you have. Unfortunately all of us also need to be able to run this test so you get to create a deployment in this unit test 😅